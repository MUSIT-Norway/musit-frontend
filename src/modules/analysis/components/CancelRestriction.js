// @flow
import React from 'react';
import { I18n } from 'react-i18nify';
import type { AppSession } from '../../../types/appSession';
import type { Restriction } from '../../../types/analysis';
import StatefulActorSuggest from './StatefulActorSuggest';

type Props = {
  appSession: AppSession,
  restriction: Restriction,
  updateRestriction: (restriction: Restriction) => void,
  clickCancel: () => void
};

export default function CancelRestriction(props: Props) {
  return (
    <div>
      <div className="form-group">
        <label className="control-label col-md-2" htmlFor="restrictedBy">
          {I18n.t('musit.analysis.restrictions.restrictionsFor')}
        </label>
        <div className="col-md-10">
          <StatefulActorSuggest
            value={props.restriction.cancelledByName}
            appSession={props.appSession}
            onChange={actorId =>
              props.updateRestriction({
                ...props.restriction,
                cancelledBy: actorId
              })}
          />
        </div>
      </div>
      <div className="form-group">
        <label className="control-label col-md-2" htmlFor="cancelCause">
          {I18n.t('musit.analysis.restrictions.reasonForCancelling')}
        </label>
        <div className="col-md-10">
          <input
            className="form-control"
            id="cancelCause"
            value={props.restriction.cancelledReason || ''}
            onChange={e =>
              props.updateRestriction({
                ...props.restriction,
                cancelledReason: e.target.value
              })}
          />
          {!props.restriction.cancelledStamp &&
            <button
              onClick={(e: any) => {
                e.preventDefault();
                props.clickCancel();
              }}
            >
              Cancel now
            </button>}

        </div>
      </div>
    </div>
  );
}
