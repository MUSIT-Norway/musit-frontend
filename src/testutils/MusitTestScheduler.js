import DiffPatcher from 'jsondiffpatch';
import { log } from 'jsondiffpatch/src/formatters/console';
import assert from 'assert';
import { TestScheduler } from 'rxjs/Rx';
import DeepDiff from 'deep-diff';

export default class MusitTestScheduler extends TestScheduler {
  constructor() {
    super();
    this.assertDeepEqual = (actual, expected) => {
      const difference = DeepDiff.diff(actual, expected);
      if (typeof difference !== 'undefined') {
        const theDiff = DiffPatcher.diff(actual, expected);
        log(theDiff);
      }
      return assert.equal(undefined, difference);
    };
  }
}
