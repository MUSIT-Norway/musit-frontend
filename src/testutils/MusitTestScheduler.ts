// @flow
import { DiffPatcher } from 'jsondiffpatch';
import { log } from 'jsondiffpatch/src/formatters/console';
import * as assert from 'assert';
import { TestScheduler } from 'rxjs';
import * as DeepDiff from 'deep-diff';
import { TODO } from '../types/common';

export default class MusitTestScheduler extends TestScheduler {
  constructor() {
    const localAssertDeepEqual = (actual: any, expected: any) => {
      const difference = DeepDiff.diff(actual, expected);
      if (typeof difference !== 'undefined') {
        const theDiff = (DiffPatcher as TODO).diff(actual, expected);
        log(theDiff);
        console.log('ACTUAL:', JSON.stringify(actual, null, 2));
        console.log('EXPECTED:', JSON.stringify(expected, null, 2));
      }
      return assert.equal(undefined, difference);
    };
    super(localAssertDeepEqual);
    this.assertDeepEqual = localAssertDeepEqual; //Not sure if this is needed
  }
}
