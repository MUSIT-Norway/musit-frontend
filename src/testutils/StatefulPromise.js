// @flow

/**
 * Usage:
 *
 * const promiseHelper = new StatefulPromise();
 * callMethodThatRequiresAFunctionThatReturnsAPromise(promiseHelper.createPromise())
 *   .then(() => {
 *     expect(promiseHelper.value).toEqual('SOme wierd stuff');
 *   });
 */
export default class StatefulPromise {
  value: any;

  createPromise() {
    return (value: any) =>
      new Promise((resolve: any => void) => {
        this.value = value;
        resolve(value);
      });
  }
}
