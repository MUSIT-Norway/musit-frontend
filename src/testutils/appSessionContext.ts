// @flow

import * as PropTypes from 'prop-types';
import { Observable } from 'rxjs';

import { AppSession } from '../types/appSession';

export const createAppSessionContext = (appSession: AppSession) => ({
  context: {
    appSession$: Observable.of(appSession)
  },
  childContextTypes: { appSession$: PropTypes.instanceOf(Observable).isRequired }
});

export const createEnLangAppSessionContext = () =>
  createAppSessionContext({
    museumId: 99,
    collectionId: '0000-0000-0000',
    accessToken: 'dummy-token',
    actor: { fn: 'Test Testersen' },
    language: { isEn: false, isNo: true },
    rolesForModules: {
      collectionManagementRead: true,
      collectionManagementWrite: true,
      storageFacilityRead: true,
      storageFacilityWrite: true,
      storageFacilityAdmin: true,
      documentArchiveRead: true,
      documentArchiveWrite: true
    }
  });

export const createNoLangAppSessionContect = () =>
  createAppSessionContext({
    museumId: 99,
    collectionId: '0000-0000-0000',
    accessToken: 'dummy-token',
    actor: { fn: 'Test Testersen' },
    language: { isEn: false, isNo: true },
    rolesForModules: {
      collectionManagementRead: true,
      collectionManagementWrite: true,
      storageFacilityRead: true,
      storageFacilityWrite: true,
      storageFacilityAdmin: true,
      documentArchiveRead: true,
      documentArchiveWrite: true
    }
  });
