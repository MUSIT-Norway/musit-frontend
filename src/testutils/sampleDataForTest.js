// @flow
import type { AppSession } from '../types/appSession';
import type { History } from '../types/Routes';
import type { AnalysisCollection, AnalysisEvent } from '../types/analysis';
import type { SampleDataExtended } from '../types/samples';
import type { MusitObject } from '../types/object';
import type { Field } from '../forms/form';
import { noMapper } from '../forms/mappers';
import {
  getStrField,
  getNumberField,
  getBoolField,
  getArrField,
  getCompositeField
} from '../forms/form';
import type { FormData as AnalysisFormData } from '../modules/analysis/shared/formType';

export const appSession: AppSession = {
  museumId: 99,
  collectionId: '1234',
  accessToken: '45667',
  actor: {
    fn: 'Test'
  },
  language: {
    isEn: false,
    isNo: true
  }
};

export const history: History = {
  push: () => {},
  replace: () => {},
  goBack: () => {},
  goForward: () => {}
};

export const analysis: AnalysisCollection = {
  analysisTypeId: 1,
  eventDate: '2017-03-16T14:37:45+00:00',
  id: 2,
  museumNo: 'MusK58',
  note: 'fdsfsd sdsa 2',
  objectId: 'adea8141-8099-4f67-bff9-ea5090e18335',
  partOf: 1,
  registeredBy: '7dcc7e82-a18c-4e2e-9d83-2b25c132fc3e',
  registeredByName: 'Rituvesh Kumar',
  registeredDate: '2017-04-03T10:36:34+00:00',
  subNo: '2',
  events: [],
  term: 'Mansjettknapp',
  type: 'Analysis'
};

export const sample: SampleDataExtended = {
  id: '123',
  uuid: '0000-0000-123',
  objectId: '0000-0000-123',
  originatedObjectUuid: '0000-0000-123',
  museumId: 99,
  status: 1,
  leftoverSample: 1,
  isExtracted: false,
  isDeleted: false,
  sampleTypeId: 1,
  museumNo: 'M1234',
  term: 'Carex saxatilis',
  subNo: 'a',
  objectType: 'collection',
  currentLocation: { pathNames: null, breadcrumb: null },
  parentObject: {
    objectId: '000-0000-0001',
    objectType: 'collection',
    sampleOrObjectData: {}
  },
  registeredDate: 'some date',
  registeredStamp: { user: '000-0000-0001', date: '2017' },
  // updatedStamp: {name: null},
  objectUUID: '0000-0000-123',
  sampleSubType: 'some sub type',
  doneBy: 'none',
  hasAnalyse: false,
  date: 'some date',
  breadcrumb: [],
  details: 'some details',
  nodeId: 'blee',
  sampleNum: 12,
  sampleType: {
    sampleTypeId: 1,
    sampleType: null,
    enSampleType: 'en',
    noSampleType: 'no',
    noSampleSubType: null,
    enSampleSubType: null
  }
};

export function createAnalysisEventWithObject(
  id: number,
  uuid: string,
  museumNo: string,
  subNo: string,
  term: string
): AnalysisEvent {
  return {
    id: 2 * id,
    affectedThing: uuid,
    affectedType: 'collection',
    analysisTypeId: 5,
    doneBy: 'some uuid',
    doneDate: 'some date',
    partOf: 3 * id,
    registeredBy: 'some uuid',
    registeredDate: 'some date',
    responsible: 'some uuid',
    type: 'Analysis',
    objectData: createObject(id, uuid, museumNo, subNo, term)
  };
}

export function createObject(
  id: number,
  uuid: string,
  museumNo: string,
  subNo: string,
  term: string
): MusitObject {
  return {
    id: id,
    uuid: uuid,
    museumId: 99,
    museumNo: museumNo,
    subNo: subNo,
    term: term,
    collection: 8,
    materials: [],
    locations: [],
    coordinates: [],
    objectType: 'collection'
  };
}

export const object: MusitObject = {
  id: 56,
  uuid: '12080e3e-2ca2-41b1-9d4a-4d72e292dcd8',
  museumId: 99,
  museumNo: 'MusN11',
  term: 'Solsikke',
  collection: 8,
  materials: [],
  locations: [],
  coordinates: [],
  objectType: 'collection'
};

export function createField<T>(
  name: string,
  value: T,
  valid?: boolean = false
): Field<T> {
  return {
    name: name,
    rawValue: noMapper.toRaw(value),
    defaultValue: value,
    status: {
      valid: valid
    },
    validator: {},
    mapper: noMapper
  };
}

export const analysisForm: AnalysisFormData = {
  id: getNumberField('', 1),
  analysisTypeId: getNumberField('', 1),
  analysisTypeCategory: getStrField('', '1'),
  doneBy: getStrField('', '1'),
  doneDate: getStrField('', '1'),
  registeredBy: getStrField('', '1'),
  registeredByName: getStrField('', '1'),
  registeredDate: getStrField('', '1'),
  responsible: getStrField('', '1'),
  administrator: getStrField('', '1'),
  persons: getArrField('', [
    {
      name: 'jarl',
      role: 'responsible'
    },
    {
      name: 'line',
      role: 'creator',
      date: '2017-06-29T07:54:22+00:00'
    }
  ]),
  completedBy: getStrField('', '1'),
  completedDate: getStrField('', '1'),
  objectId: getStrField('', '1'),
  note: getStrField('', '1'),
  type: getStrField('', '1'),
  partOf: getStrField('', '1'),
  result: getStrField('', '1'),
  orgId: getNumberField('orgId', 1),
  externalSource: getStrField('', '1'),
  comments: getStrField('', '1'),
  reason: getStrField('', '1'),
  caseNumbers: getArrField('', []),
  completeAnalysis: getStrField('', '1'),
  // restriction
  restrictions: getBoolField('', true),
  restriction: getCompositeField('', {
    requester: '3cbf15cb-8348-4e66-99a4-bc314da57444',
    requesterName: 'Jarl',
    reason: 'fin Ã¥rsak',
    caseNumbers: ['dddd', '44555', '55555'],
    expirationDate: '2017-03-29T07:54:22+00:00',
    cancelledReason: 'meh'
  }),
  // object details
  museumNo: getStrField('', '1'),
  subNo: getStrField('', '1'),
  term: getStrField('', '1'),
  eventDate: getStrField('', '1'),
  actor: getStrField('', '1'),
  events: getArrField('', []),
  updatedBy: getStrField('', '1'),
  updatedByName: getStrField('', '1'),
  updatedDate: getStrField('', '1'),
  status: getNumberField('status', 1)
};
