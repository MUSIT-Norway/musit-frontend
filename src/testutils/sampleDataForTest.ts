// @flow
import { AppSession } from "../types/appSession";
import {
  History,
  LocationListener,
  LocationDescriptorObject,
  Action,
  UnregisterCallback
} from "history";
import { AnalysisCollection, AnalysisEvent } from "../types/analysis";
import { SampleDataExtended } from "../types/samples";
import { MusitObject } from "../types/object";
import { Field } from "../forms/form";
import { noMapper } from "../forms/mappers";
import {
  getStrField,
  getNumberField,
  getBoolField,
  getArrField,
  getCompositeField
} from "../forms/form";
import { FormData as AnalysisFormData } from "../modules/analysis/shared/formType";

export const appSession: AppSession = {
  museumId: 99,
  collectionId: "1234",
  accessToken: "45667",
  actor: {
    fn: "Test"
  },
  language: {
    isEn: false,
    isNo: true
  },
  rolesForModules: {
    collectionManagementRead: true,
    collectionManagementWrite: true,
    storageFacilityRead: true,
    storageFacilityWrite: true,
    storageFacilityAdmin: true,
    documentArchiveRead: true,
    documentArchiveWrite: true
  }
};

export const history: History = {
  push: () => {},
  replace: () => {},
  goBack: () => {},
  goForward: () => {},
  length: 0,
  action: (undefined as any) as Action,
  location: undefined as any, //Can cast it as Location, but we have two different Locations in scope here, so I didn't bother
  go: (n: number) => {},
  block: (prompt?: any) => (undefined as any) as UnregisterCallback,
  listen: (listener: LocationListener) => (undefined as any) as UnregisterCallback,
  createHref: (location: LocationDescriptorObject) => (undefined as any) as string
};

export const analysis: AnalysisCollection = {
  id: 1,
  analysisTypeId: 27,
  doneBy: "7dcc7e82-a18c-4e2e-9d83-2b25c132fc3e",
  doneDate: "2018-06-12T11:13:19+00:00",
  registeredBy: "7dcc7e82-a18c-4e2e-9d83-2b25c132fc3e",
  registeredDate: "2018-06-12T11:14:43+00:00",
  responsible: "7dcc7e82-a18c-4e2e-9d83-2b25c132fc3e",
  administrator: "7dcc7e82-a18c-4e2e-9d83-2b25c132fc3e",
  completedBy: "7dcc7e82-a18c-4e2e-9d83-2b25c132fc3e",
  completedDate: "2018-06-12T11:13:43+00:00",
  note: "test",
  result: {
    registeredBy: "7dcc7e82-a18c-4e2e-9d83-2b25c132fc3e",
    registeredDate: "2018-06-12T11:14:44+00:00",
    extRef: ["bcvbvbvc"],
    comment: "vcvcvx",
    attachments: ["7b0aac35-3bc4-452f-9f8d-8964a91fdcf0"],
    type: "GenericResult"
  },
  events: [
    {
      objectData: {
        id: 14,
        uuid: "a982bcc0-133c-4af8-99cc-f02df98c2b0b",
        museumId: 99,
        museumNo: "MusK23",
        term: "Lite skaft av ben",
        collection: 1,
        materials: [],
        locations: [],
        coordinates: [],
        isDeleted: false,
        objectType: "collection"
      },
      id: 2,
      analysisTypeId: 27,
      doneBy: "7dcc7e82-a18c-4e2e-9d83-2b25c132fc3e",
      doneDate: "2018-06-12T11:13:19+00:00",
      registeredBy: "7dcc7e82-a18c-4e2e-9d83-2b25c132fc3e",
      registeredDate: "2018-06-12T11:14:43+00:00",
      responsible: "7dcc7e82-a18c-4e2e-9d83-2b25c132fc3e",
      //TODO: administrator: "7dcc7e82-a18c-4e2e-9d83-2b25c132fc3e",
      //TODO: completedBy: "7dcc7e82-a18c-4e2e-9d83-2b25c132fc3e",
      //TODO: completedDate: "2018-06-12T11:13:43+00:00",
      affectedThing: "a982bcc0-133c-4af8-99cc-f02df98c2b0b",
      affectedType: "collection",
      partOf: 1,
      note: "test",
      type: "Analysis"
    }
  ],
  restriction: {
    requester: "7dcc7e82-a18c-4e2e-9d83-2b25c132fc3e",
    expirationDate: "2018-06-13T22:00:00+00:00",
    reason: "dfdfs",
    caseNumbers: ["123"],
    registeredStamp: {
      user: "7dcc7e82-a18c-4e2e-9d83-2b25c132fc3e",
      date: "2018-06-12T11:14:43+00:00"
    },
    requesterName: "Rituvesh Kumar"
  },
  reason: "Dating",
  status: 1,
  caseNumbers: ["123"],
  orgId: 389,
  type: "AnalysisCollection",
  registeredByName: "Rituvesh Kumar",
  doneByName: "Rituvesh Kumar",
  responsibleName: "Rituvesh Kumar",
  administratorName: "Rituvesh Kumar",
  completedByName: "Rituvesh Kumar"
  /*TODO: Include this, I removed it in order to get stuff to compile. (Rituvesh and I added this much larger object actually used in the app earlier today, perhaps a bit too optimistic)
  files: [
    {
      id: "cf70910e-6e31-11e8-9251-13f2b8765574",
      fid: "7b0aac35-3bc4-452f-9f8d-8964a91fdcf0",
      title: "Screenshot from 2017-05-31 11-28-54 (13th copy).png",
      size: "305586",
      fileType: "image/png",
      owner: {
        ownerId: "99",
        ownerType: "org"
      },
      path: "/root/Modules/Analysis/1",
      version: 1,
      published: false,
      createdStamp: {
        date: "2018-06-12T11:14:44.649+00:00",
        by: "7dcc7e82-a18c-4e2e-9d83-2b25c132fc3e"
      },
      documentDetails: {
        number: 1
      },
      type: "archive_document"
    }
    
  ]*/
};

export const sample: SampleDataExtended = {
  id: "123",
  uuid: "0000-0000-123", //TODO: Is uuid needed?
  objectId: "0000-0000-123",
  originatedObjectUuid: "0000-0000-123",
  museumId: 99,
  status: 1,
  leftoverSample: 1,
  isExtracted: false,
  isDeleted: false,
  sampleTypeId: 1,
  museumNo: "M1234",
  term: "Carex saxatilis",
  subNo: "a",
  objectType: "collection",
  currentLocation: { pathNames: null, breadcrumb: null },
  parentObject: {
    objectId: "000-0000-0001",
    objectType: "collection",
    sampleOrObjectData: {}
  },
  registeredDate: "some date",
  registeredStamp: { user: "000-0000-0001", date: "2017" },
  // updatedStamp: {name: null},
  objectUUID: "0000-0000-123",
  sampleSubType: "some sub type",
  doneBy: "none",
  hasAnalyse: false,
  date: "some date",
  breadcrumb: [],
  details: "some details",
  nodeId: "blee",
  sampleNum: 12,
  sampleType: {
    sampleTypeId: 1,
    enSampleType: "en",
    noSampleType: "no",
    noSampleSubType: "no sub",
    enSampleSubType: "en sub"
  }
};

export function createAnalysisEventWithObject(
  id: number,
  uuid: string,
  museumNo: string,
  subNo: string,
  term: string
): AnalysisEvent {
  return {
    id: 2 * id,
    affectedThing: uuid,
    affectedType: "collection",
    analysisTypeId: 5,
    doneBy: "some uuid",
    doneDate: "some date",
    partOf: 3 * id,
    registeredBy: "some uuid",
    registeredDate: "some date",
    responsible: "some uuid",
    type: "Analysis",
    objectData: createObject(id, uuid, museumNo, subNo, term)
  };
}

export function createObject(
  id: number,
  uuid: string,
  museumNo: string,
  subNo: string,
  term: string
): MusitObject {
  return {
    id: id,
    uuid: uuid,
    museumId: 99,
    museumNo: museumNo,
    subNo: subNo,
    term: term,
    collection: 8,
    materials: [],
    locations: [],
    coordinates: [],
    objectType: "collection"
  };
}

export const object: MusitObject = {
  id: 56,
  uuid: "12080e3e-2ca2-41b1-9d4a-4d72e292dcd8",
  museumId: 99,
  museumNo: "MusN11",
  term: "Solsikke",
  collection: 8,
  materials: [],
  locations: [],
  coordinates: [],
  objectType: "collection"
};

export function createField<T>(name: string, value: T, valid: boolean = false): Field<T> {
  return {
    name: name,
    rawValue: noMapper.toRaw(value),
    defaultValue: value,
    status: {
      valid: valid
    },
    validator: {},
    mapper: noMapper
  };
}

export const analysisForm: AnalysisFormData = {
  id: getNumberField("", 1),
  analysisTypeId: getNumberField("", 1),
  analysisTypeCategory: getStrField("", "1"),
  doneBy: getStrField("", "1"),
  doneDate: getStrField("", "1"),
  registeredBy: getStrField("", "1"),
  registeredByName: getStrField("", "1"),
  registeredDate: getStrField("", "1"),
  responsible: getStrField("", "1"),
  resultFiles: getArrField("", []),
  administrator: getStrField("", "1"),
  persons: getArrField("", [
    {
      name: "jarl",
      role: "responsible"
    },
    {
      name: "line",
      role: "creator",
      date: "2017-06-29T07:54:22+00:00"
    }
  ]),
  completedBy: getStrField("", "1"),
  completedDate: getStrField("", "1"),
  objectId: getStrField("", "1"),
  note: getStrField("", "1"),
  type: getStrField("", "1"),
  partOf: getStrField("", "1"),
  result: getCompositeField("", null),
  orgId: getNumberField("orgId", 1),
  externalSource: getStrField("", "1"),
  comments: getStrField("", "1"),
  reason: getStrField("", "1"),
  caseNumbers: getArrField("", []),
  completeAnalysis: getStrField("", "1"),
  // restriction
  restrictions: getBoolField("", true),
  restriction: getCompositeField("", {
    requester: "3cbf15cb-8348-4e66-99a4-bc314da57444",
    requesterName: "Jarl",
    reason: "fin Ã¥rsak",
    caseNumbers: ["dddd", "44555", "55555"],
    expirationDate: "2017-03-29T07:54:22+00:00",
    cancelledReason: "meh"
  }),
  // object details
  museumNo: getStrField("", "1"),
  subNo: getStrField("", "1"),
  term: getStrField("", "1"),
  eventDate: getStrField("", "1"),
  actor: getStrField("", "1"),
  events: getArrField("", []),
  updatedBy: getStrField("", "1"),
  updatedByName: getStrField("", "1"),
  updatedDate: getStrField("", "1"),
  status: getNumberField("status", 1)
};
