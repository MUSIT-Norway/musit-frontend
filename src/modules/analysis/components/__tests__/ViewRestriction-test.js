// @flow
import React from 'react';
import ViewRestrictionComponent, { ViewRestriction } from '../ViewRestriction';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';
import { appSession } from '../../../../testutils/sampleDataForTest';
import CancelRestriction from '../CancelRestriction';

declare var describe: any;
declare var it: any;
declare var expect: any;

const restriction = {
  reason: 'Initial reason',
  expirationDate: '2017-08-21T08:17:03+00:00',
  caseNumbers: ['This', 'is', 'not', 'it'],
  requester: 'a50da1c5-9359-4b42-b94a-8faa04ff95cd',
  requesterName: 'dummy user',
  registeredStamp: {
    user: 'a50da1c5-9359-4b42-b94a-8faa04ff95cd',
    date: '2017-08-21T08:17:03+00:00'
  }
};

describe('ViewRestrictionComponent', () => {
  const clickCancel = sinon.spy();
  const updateRestriction = sinon.spy();
  const cancelRestriction = sinon.spy();
  const toggleCancelDialog = sinon.spy();

  it('should render restriction', () => {
    const showCancelDialog = false;

    const Comp = shallow(
      <ViewRestrictionComponent
        clickCancel={clickCancel}
        appSession={appSession}
        restriction={restriction}
        updateRestriction={updateRestriction}
        cancelRestriction={cancelRestriction}
        showCancelDialog={showCancelDialog}
        toggleCancelDialog={toggleCancelDialog}
        isRestrictionValidForCancellation={false}
      />
    );

    expect(Comp.find(ViewRestriction).length).toBe(1);
  });

  it('should render cancel dialog', () => {
    const showCancelDialog = true;

    const Comp = shallow(
      <ViewRestrictionComponent
        clickCancel={clickCancel}
        appSession={appSession}
        restriction={restriction}
        updateRestriction={updateRestriction}
        cancelRestriction={cancelRestriction}
        showCancelDialog={showCancelDialog}
        toggleCancelDialog={toggleCancelDialog}
        isRestrictionValidForCancellation={false}
      />
    );

    expect(Comp.find(ViewRestriction).length).toBe(0);
    expect(Comp.find(CancelRestriction).length).toBe(1);
    Comp.find(CancelRestriction).props().clickCancel();
    expect(cancelRestriction.calledOnce).toBe(true);
    expect(toggleCancelDialog.calledOnce).toBe(true);
  });
});

describe('ViewRestriction', () => {
  it('should render restriction', () => {
    const clickCancel = sinon.spy();

    const Comp = shallow(
      <ViewRestriction
        clickCancel={clickCancel}
        appSession={appSession}
        restriction={restriction}
      />
    );

    expect(shallowToJson(Comp)).toMatchSnapshot();
  });

  it('should call clickCancel when button is clicked', () => {
    const clickCancel = sinon.spy();
    const preventDefault = sinon.spy();

    const Comp = shallow(
      <ViewRestriction
        clickCancel={clickCancel}
        appSession={appSession}
        restriction={restriction}
      />
    );

    Comp.find('button').simulate('click', { preventDefault });
    expect(clickCancel.calledOnce).toBe(true);
    expect(preventDefault.calledOnce).toBe(true);
  });
});
