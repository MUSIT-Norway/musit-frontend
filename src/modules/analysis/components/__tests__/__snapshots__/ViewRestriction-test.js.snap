// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewRestriction should render restriction 1`] = `
<div>
  <div
    className="form-group"
  >
    <label
      className="control-label col-md-2"
      htmlFor="restrictedBy"
    >
      Klausulert for
    </label>
    <div
      className="col-md-10"
    >
      <p
        className="form-control-static"
      >
        dummy user
      </p>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className="control-label col-md-2"
      htmlFor="restrictionCause"
    >
      Ã…rsak til klausulering
    </label>
    <div
      className="col-md-10"
    >
      <p
        className="form-control-static"
      >
        Initial reason
      </p>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className="control-label col-md-2"
      htmlFor="restrictionCaseNumbers"
    >
      Saksnummer
    </label>
    <div
      className="col-md-5"
    >
      <p
        className="form-control-static"
      >
        This, is, not, it
      </p>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      className="control-label col-md-2"
      htmlFor="restrictionExpirationEndDate"
    >
      Sluttdato
    </label>
    <div
      className="col-md-5"
    >
      <p
        className="form-control-static"
      >
        21.08.2017
      </p>
      <button
        onClick={[Function]}
      >
        Click me please
      </button>
    </div>
  </div>
</div>
`;
